app/jobs/base.rb

```diff
@@ -461,6 +461,45 @@ def self.enqueue_at(datetime, job_name, opts = {})
     enqueue_in(secs, job_name, opts)
   end
 
+  @@enqueue_after = []
+
+  def self.process_enqueue_after
+    @@enqueue_after.filter! do |job_ids, job_name, opts|
+      pending_job_ids = Set.new
+
+      Sidekiq::Queue.all.each do |queue|
+        queue.each { |job| pending_job_ids << job.jid if job_ids.include?(job.jid) }
+      end
```

Comment:
I don't think we can rely on `Sidekiq::Queue` because the number of jobs in the queue may be huge. Do have a read through https://www.mikeperham.com/2021/04/20/a-tour-of-the-sidekiq-api/ for more information. 

Can you see if we can use `Sidekiq::ScheduledSet.new` or `Sidekiq::RetrySet.new` to scan for the relevant jobs instead? That should be way more efficient and accurate.